adapters/flashbots/signature.go

```diff
@@ -0,0 +1,67 @@
+// Package flashbots provides methods for parsing the X-Flashbots-Signature header.
+package flashbots
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/ethereum/go-ethereum/accounts"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/crypto"
+)
+
+var (
+	ErrNoSignature      = errors.New("no signature provided")
+	ErrInvalidSignature = errors.New("invalid signature provided")
+)
+
+func ParseSignature(header string, body []byte) (string, error) {
+	if header == "" {
+		return "", ErrNoSignature
+	}
+
+	splitSig := strings.Split(header, ":")
+	if len(splitSig) != 2 {
+		return "", ErrInvalidSignature
+	}
+
+	return VerifySignature(body, splitSig[0], splitSig[1])
+}
+
+func VerifySignature(body []byte, signingAddressStr, signatureStr string) (string, error) {
```

Comment:
explicit names for the return values would increase clarity 

```suggestion
func VerifySignature(body []byte, signingAddressStr, signatureStr string) (signaturePubkeyHex string, err error) {
```

Reply:
Add in https://github.com/flashbots/rpc-endpoint/pull/151/commits/d90fb9b1c8de38358dbf4e5c748e52b234462300, still need to write unit tests those are coming next.